src/services/job-role-service.ts

```diff
@@ -0,0 +1,89 @@
+import type { JobRole } from "../models/job-role.js";
+import type { JobRoleService } from "./interfaces.js";
+
+export class JobRoleServiceImpl implements JobRoleService {
```

Comment:
while this name technically works, it does not explain that this particular implementation uses an array stored in memory. Remember there will be another one soon that actually calls the API